### ISSUE TITLE
Toro goes Blank Screen After Fragment Shifting

### ISSUE THREAD

**Issue by harikautilya** [eneim/toro#234]
I Have the system of my app as follows 
Fragment A ->FragmentA1(Toro players in recycler view),FragmentA2(Toro players in recycler view)
Fragment B-> Information Page
Fragment C -> information Study

ViewPager on activity which has Fragement A,B,C
ViewPager on Fragment A which has A1,A2

I am currently getting two issues
The issue 1 is when i start the app both the fragments A1,A2 starts playing.
The issue 2 is when i move from fragment C to A the entire Toro gets Blank like a Black screen

I tired to find the issue by debugging and found at this point it is always returning null at Toro.unregister().
 
if (manager.getPlayer() != null) 

hence the players are not pausing.
I am calling the Toro.register as soon as i set the adpater which implements the player manger as per guidelines.

i am kinda confused. Been 2 days and yet no progess .
Any help is appreciated .
Thank you for the awesome library.

**Comment by eneim** [comment]
@harikautilya Can you give Toro 3 a try? If you are at the starting point of your implementation. The Toro v2 could not smoothly support ViewPager as I mentioned in the release note of 2.3.0, so sorry for that.

**Comment by harikautilya** [comment]
I have completed finished my app the only thing left out is this issue. Migrating it will cause me to change the entire system. Is there any work around please let me in the mean while i am going through the new version of toro. Thank you for your patience

**Comment by eneim** [comment]
@harikautilya So sorry to hear that. I have tried to make it supports ViewPager well in 2.x but it turns out that current design is pretty tricky, which makes me to decide to have a rewrite for toro. At the moment, what I can think of is to register/unregister the RecyclerView those are on ViewPager (so: if your ViewPager contains FragmentA(RecyclerViewA), FragmentB(RecyclerViewB), FragmentC(RecyclerViewC), when FragmentA is shown you can remove the RecyclerView on FragmentB and FragmentC and so on...)

**Comment by harikautilya** [comment]
@eneim  yes i have proceeded like that for the scene But the issue is all the fragments starts playing coz before the register could happen the un-register is called as per activity life-cycle of the android fragment.
So i decided to wait for a stable version of the toro that supports a view pager.

In mean while maybe i will experiment a little with a custom code using toro as basic import classes.
Thank you . Eagerly waiting

**Comment by eneim** [comment]
@harikautilya thank you. at the mean time you can checkout latest code of v3. the demo App has a beautiful (from my point of view) demonstration of using ViewPager. there still some tricky steps when using setUserVisibility, but so far so good it looks way better. please take a look if it helps. I'm just waiting for Android team to release their last preview before official launch to push my 3.0.0 out. please wait a bit.

**Comment by harikautilya** [comment]
@eneim 
Sure i will definetly have a look
